src/org/opensolaris/opengrok/web/api/error/GenericExceptionMapper.java

```diff
@@ -22,20 +22,24 @@
  */
 package org.opensolaris.opengrok.web.api.error;
 
-import javax.ws.rs.core.MediaType;
+import org.opensolaris.opengrok.logger.LoggerFactory;
+
 import javax.ws.rs.core.Response;
 import javax.ws.rs.ext.ExceptionMapper;
 import javax.ws.rs.ext.Provider;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
 @Provider
 public class GenericExceptionMapper implements ExceptionMapper<Exception> {
 
+    private static final Logger logger = LoggerFactory.getLogger(GenericExceptionMapper.class);
+
     @Override
     public Response toResponse(final Exception e) {
-        return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
-                .entity(e.getMessage())
-                .type(MediaType.TEXT_PLAIN)
-                .build();
+        logger.log(Level.FINE, "Exception while processing request", e);
```

Comment:
Yes, it does. I thought `FINE` was a better option because it is not known if the error is serious or not... we can get here almost from any exception and we don't know that the webapp has some serious problems (which `WARNING` or `SEVERE`) indicate.

Reply:
but with this any exception in the api is logged with just `FINE`?

Reply:
The one that is not caught, yes. IMHO if some serious problems arose during the execution then the programmer should not rely on the underlying implementation and throw exceptions out but log them properly.

Reply:
But this aborts the request with 500 and that doesn't sounds `FINE` to me. I'm for at least `WARNING` if not more.

Reply:
It's up to the admin to determine wheter the exception was worth logging/fixing/doing something else or if it was just a trash.

Reply:
But then if one endpoint which is accessed very often would have a silly mistake then all the logs will be swamped with `WARNING` logs which is not good either.

Reply:
But that's a job for automated tools, server checks to discover the errors, inform the admin and then he can take the action.

If anything is broken, then someone should be notified.

Reply:
Okay, already changed it :)